pkg/scalers/elasticsearch_scaler.go

```diff
@@ -0,0 +1,277 @@
+package scalers
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/elastic/go-elasticsearch/v7"
+	kedautil "github.com/kedacore/keda/v2/pkg/util"
+	"github.com/tidwall/gjson"
+	"k8s.io/api/autoscaling/v2beta2"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/metrics/pkg/apis/external_metrics"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type elasticsearchScaler struct {
+	metadata *elasticsearchMetadata
+	esClient *elasticsearch.Client
+}
+
+type elasticsearchMetadata struct {
+	addresses          []string
+	unsafeSsl          bool
+	username           string
+	password           string
+	indexes            []string
+	searchTemplateName string
+	parameters         []string
+	valueLocation      string
+	targetValue        int
+}
+
+var elasticsearchLog = logf.Log.WithName("elasticsearch_scaler")
+
+// NewElasticsearchScaler creates a new elasticsearch scaler
+func NewElasticsearchScaler(config *ScalerConfig) (Scaler, error) {
+	meta, err := parseElasticsearchMetadata(config)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing elasticsearch metadata: %s", err)
+	}
+
+	esClient, err := newElasticsearchClient(meta)
+	if err != nil {
+		return nil, fmt.Errorf("error getting elasticsearch client: %s", err)
+	}
+	return &elasticsearchScaler{
+		metadata: meta,
+		esClient: esClient,
+	}, nil
+}
+
+const defaultUnsafeSsl = false
+
+func parseElasticsearchMetadata(config *ScalerConfig) (*elasticsearchMetadata, error) {
+	meta := elasticsearchMetadata{}
+
+	var err error
+	addresses, err := GetFromAuthOrMeta(config, "addresses")
+	if err != nil {
+		return nil, err
+	}
+	meta.addresses = splitAndTrimBySep(addresses, ",")
+
+	if val, ok := config.TriggerMetadata["unsafeSsl"]; ok {
+		meta.unsafeSsl, err = strconv.ParseBool(val)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing unsafeSsL: %s", err)
```

Comment:
```suggestion
			return nil, fmt.Errorf("error parsing unsafeSsl: %s", err)
```
Does the last L need to be in capital letter?